tools/config_parser.c

```diff
@@ -694,8 +697,10 @@ static int __cp_parse_smbconfig(const char *smbconf, GHFunc cb,
 		goto out;
 
 	global_group = g_hash_table_lookup(parser.groups, "global");
-	if (global_group)
-		global_group->cb_mode = cb_mode;
+	if (!global_group)
+		goto out;
```

Comment:
What prompted this change? You are skipping config file parsing altogether if there is no global section:
```sh
cat <<'EOF' >test1.conf
[MyShare]
        path = /home/atte/MyShare
EOF
sudo sh -c "ksmbd.mountd -v -n2 -c test1.conf -u '' &"; sleep 5; sudo ksmbd.control -s
# [ksmbd-worker/40138]: DEBUG: Can't open `': No such file or directory
# [ksmbd-worker/40138]: INFO: User database does not exist, only guest sessions may work
# [ksmbd-worker/40138]: ERROR: nl_recv() returned error code -7: Invalid input data or parameter
# [ksmbd.mountd/40137]: INFO: Worker PID 40138 changed state
# ...

cat <<'EOF' >test2.conf
[global]
        comment = My Default Comment
[MyShare]
        path = /home/atte/MyShare
EOF
sudo sh -c "ksmbd.mountd -v -n2 -c test2.conf -u '' &"; sleep 5; sudo ksmbd.control -s
# [ksmbd-worker/40158]: DEBUG: Can't open `': No such file or directory
# [ksmbd-worker/40158]: INFO: User database does not exist, only guest sessions may work
# [ksmbd-worker/40158]: DEBUG: New share `IPC$'
# [ksmbd-worker/40158]: DEBUG: New share `MyShare'
# [ksmbd-worker/40158]: DEBUG: New user `nobody'
# ...
```

Reply:
I wasn't even aware it was legitimate to not have a global section ...

I received reports that  mountd crashed. Since the reports didn't come with the configuration files, I tried to isolate the config parsing code in order to fuzz that part of the code. Which brought about the proposed fixes.

For what it's worth, the invalid configuration looked like this (key/value pair outside of any section):
```

key = value
[section]
key1 = value1
```

Reply:
Yes, it is fine to not have a global section just as it is fine to not define every global parameter within it.

Good catch regarding out-of-group key-value definitions being a problem, i.e. them not being ignored.